environment-dev.yml

```diff
@@ -21,7 +21,7 @@ dependencies:
   - astropy
   - click>8.0
   - cython
-  - numpy>1.20
+  - numpy>1.20,<2.2
```

Comment:
The requirement can now be <=2.2 right ?

Also regarding should we impose an < to the python version as well ? (If numba lags behind 1 python version)

Reply:
I wouldn't do that. `numba` correctly pins the numpy and python versions. You, as in gammapy itself, doesn't have this requirement. 

Since you made the dependencies on numba even optional, it makes even less sense to restrict required dependencies based on what an optional dependency supports.

pip will resolve correctly when doing `pip install gammapy` what versions are compatible